app/upload_test.go

```diff
@@ -300,7 +301,7 @@ func TestUploadDataConcurrent(t *testing.T) {
 	wg.Wait()
 
 	// Verify that only 1 request was able to finish the upload.
-	require.Equal(t, int32(n*2-2), nErrs)
+	require.Equal(t, int32(n-1), nErrs)
```

Comment:
This is not very clear to me. We are still doing (n-1) + (n-1) right? How does it make a difference if we set the variable to 0 in between?

Reply:
The checks are the mathematically the same. Since we want only one call to go through and the rest to error out  (n-1 errors) I actually thought not resetting the counter was making things **less** clear as we had to account for the errors from a previous step, hence the change.